app/models/badge.rb

```diff
@@ -329,12 +329,38 @@ def self.ensure_consistency!
     Badge.find_each(&:reset_grant_count!)
   end
 
+  def display_name
+    if self.system?
+      key = "admin_js.badges.badge.#{i18n_name}.name"
+      I18n.t(key, default: self.name)
+    else
+      self.name
+    end
+  end
+
+  def long_description
+    if self[:long_description].present?
+      self[:long_description]
+    else
+      key = "badges.long_descriptions.#{i18n_name}"
+      I18n.t(key, default: '')
+    end
+  end
+
+  def slug
+    Slug.for(self.display_name, '-')
```

Comment:
We always need a slug for badges. If there isn't one, `- ` will be used as slug.
URLs without a slug like https://meta.discourse.org/badges/2 do not work when entered in the browser's address bar. I'm not sure if and how this can be changed so that badge URLs without slug work.